openhands/core/config/sandbox_config.py

```
@@ -74,3 +76,26 @@ class SandboxConfig(BaseModel):
     selected_repo: str | None = Field(default=None)
 
     model_config = {'extra': 'forbid'}
+
+    @classmethod
+    def from_toml_section(cls, data: dict) -> dict[str, 'SandboxConfig']:
+        """
+        Create a mapping of SandboxConfig instances from a toml dictionary representing the [sandbox] section.
+
+        The configuration is built from all keys in data.
+
+        Returns:
+            dict[str, SandboxConfig]: A mapping where the key "sandbox" corresponds to the [sandbox] configuration
+        """
+        # Initialize the result mapping
+        sandbox_mapping: dict[str, SandboxConfig] = {}
+
+        # Try to create the base config
+        try:
+            sandbox_mapping['sandbox'] = cls.model_validate(data)
+        except ValidationError as e:
+            logger.warning(f'Invalid sandbox configuration: {e}. Using defaults.')
```

I wonder if we should fail hard here?

---

We could... 🤔 Current behavior is that it falls back on defaults, and the defaults are always good enough to run, but it's up to us which way we want.

The little annoying thing about hard failures is that we also have `forbid` for pydantic to reject - and give a validation error - for fields it doesn't recognize. Which means that during development, when we add an attribute on a branch, switching branches will make it crash, so it forces us to change config back and forth.

On the other hand, I guess there's a risk we'll run with something unwanted and not notice, if we don't fail?

---

Yeah my primary concern is probably the latter... 🤔 But maybe we can live with it because docker / cloud launch shouldn't require us to touch config.toml at all

---

Fixed!

Yes, it's for developing, and evals don't use it either. I guess it may be worth thinking about next time we get to it, but I changed it now. It's not the first time you expressed this kind of concern ❤️

---

Maybe hard crash always by default, and have an undocumented env var to change the behavior to lenient would be an idea here... Lenient mode (yolo mode, no nagging, just go) would be only for development really.

After all, even if people had access to the toml (which I hope they will actually), does it really help them if the values they set are ignored? Except if they're developing.